File: Challenges/4kyu/range_extraction/javascript/solution.js
function solution(list){
  let ranges = []
  let range = [list[0], list[0]]
  
  for (let i = 1; i < list.length; i++) {
    const diff = list[i] - list[i - 1] 

    if (diff === 1) range[1] = list[i]
    else if (diff > 1) {
      ranges.push(getRangeStr(range))
      range = [list[i], list[i]]
    }
  }
  
  ranges.push(getRangeStr(range))
  
  return ranges.join(",")
}

function getRangeStr(range) {
  const diff = range[1] - range[0]
  
  if (diff === 0) return "" + range[0]
  if (diff === 1) return range.join(",")
  
  return range.join("-")
}

File: Challenges/4kyu/range_extraction/python/solution.py
def solution(args):
    scopes = []
    scope = [args[0], args[0]]
    
    for i in range(1, len(args)):
        diff = args[i] - args[i - 1]

        if diff == 1: scope[1] = args[i]
        elif diff > 1:
            scopes.append(getRangeStr(scope))
            scope = [args[i], args[i]]

    scopes.append(getRangeStr(scope))
    
    return ",".join(str(n) for n in scopes)

def getRangeStr(scope):
    diff = scope[1] - scope[0]
    
    if diff == 0: return str(scope[0])
    if diff == 1: return ",".join(str(n) for n in scope)

    return "-".join(str(n) for n in scope)